zaas-client/src/main/java/org/zowe/apiml/zaasclient/service/internal/ZaasJwtService.java

```diff
@@ -66,40 +63,52 @@ public ZaasJwtService(CloseableClientProvider client, String baseUrl, ConfigProp
 
     @Override
     public String login(String userId, char[] password, char[] newPassword) throws ZaasClientException {
-        return (String) doRequest(
+        return (String) doLoginRequest(
             () -> loginWithCredentials(userId, password, newPassword),
+            this::processJwtTokenResponse,
             this::extractToken);
     }
 
     @Override
     public String login(String userId, char[] password) throws ZaasClientException {
-        return (String) doRequest(
+        return (String) doLoginRequest(
             () -> loginWithCredentials(userId, password, null),
+            this::processJwtTokenResponse,
             this::extractToken);
     }
 
-    private ClientWithResponse loginWithCredentials(String userId, char[] password, char[] newPassword) throws ZaasConfigurationException, IOException {
-        var client = httpClientProvider.getHttpClient();
+    private ClassicHttpRequest loginWithCredentials(String userId, char[] password, char[] newPassword) throws IOException {
+
         var httpPost = new HttpPost(loginEndpoint);
         String json = objectMapper.writeValueAsString(new Credentials(userId, password, newPassword));
         var entity = new StringEntity(json);
         httpPost.setEntity(entity);
         httpPost.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
-        return new ClientWithResponse(client, client.execute(httpPost));
+        return httpPost;
+    }
+
+    private SimpleHttpResponse processJwtTokenResponse(ClassicHttpResponse response) throws ParseException, IOException {
+        var cookieHeader = response.getHeaders(HttpHeaders.SET_COOKIE);
+        Map<String, Header> headers = (cookieHeader != null) ? Map.of(HttpHeaders.SET_COOKIE, cookieHeader[0]) : Map.of();
+        if (response.getEntity() != null) {
+            return new SimpleHttpResponse(response.getCode(), EntityUtils.toString(response.getEntity()), headers);
+        } else {
+            return new SimpleHttpResponse(response.getCode(), headers);
+        }
```

Comment:
```suggestion
        return new SimpleHttpResponse(response.getCode(), EntityUtils.toString(response.getEntity()), headers);
```

Reply:
EntityUtils.toString will throw an NPE if the entity is null.